android-benchmark/src/androidTest/java/CryptoBenchmark.kt

```diff
@@ -0,0 +1,316 @@
+
+  val client: Client
+  val publicSigningKey: String
+  init {
+    val (c, k) = registerClient()
+    client = c
+    publicSigningKey = k
+  }
+
+  val recordType = UUID.randomUUID().toString()
+  val key = createKey(client, recordType)
+
+  val blackhole = BlackHole()
+  private val TAG = "CryptoBenchmark"
+
+  lateinit var bytes : String
+  lateinit var record: Record
+
+  @BeforeExperiment
+  open fun setup() {
+    bytes =  randomBytes(1)!!
+    record = client.encryptRecord(recordType, RecordData(mapOf(Pair("data", bytes))), null, key)
+  }
+
+  @Benchmark
+  open fun decrypt(reps: Int): BlackHole = blackhole.apply {
+    for(i in 0 until reps)
+      consume(client.decryptExisting(record, key))
+  }
+
+  @Benchmark
+  open fun encrypt(reps: Int): BlackHole = blackhole.apply {
+    for(i in 0 until reps)
+      consume(client.encryptRecord(recordType, RecordData(mapOf(Pair("data", bytes))), null, key))
+  }
+
+  @Benchmark
+  open fun sign(reps: Int): BlackHole = blackhole.apply {
+    val localRecord = LocalRecord(mapOf(Pair("data", bytes)), LocalMeta(client.clientId(), client.clientId(), recordType, null))
+    for(i in 0 until reps)
+      consume(client.sign(localRecord))
+  }
+
+  @Benchmark
+  open fun verify(reps: Int): BlackHole = blackhole.apply {
+    val localRecord = LocalRecord(mapOf(Pair("data", bytes)), LocalMeta(client.clientId(), client.clientId(), recordType, null))
+    val signed = client.sign(localRecord)
+    for(i in 0 until reps)
+      consume(client.verify(signed, publicSigningKey))
+  }
+}
+
+@RunWith(SpannerRunner::class)
+open class CryptoBenchmark1KB {
+
+  @BenchmarkConfiguration
+  @JvmField
+  var configuration = Config.config(javaClass, 5, TimeUnit.SECONDS)
+
+  val client: Client
+  val publicSigningKey: String
+  init {
+    val (c, k) = registerClient()
+    client = c
+    publicSigningKey = k
+  }
+
+  val recordType = UUID.randomUUID().toString()
+  val key = createKey(client, recordType)
+
+  val blackhole = BlackHole()
+  private val TAG = "CryptoBenchmark"
+
+  var bytes: String? = null
+  var record: Record? = null
+
+  @BeforeExperiment
+  open fun setup() {
+    bytes =  randomBytes(1_000)!!
+    record = client.encryptRecord(recordType, RecordData(mapOf(Pair("data", bytes))), null, key)
+  }
+
+  @AfterExperiment
+  open fun teardown() {
+    bytes = null
+    record = null
+  }
+
+  @Benchmark
+  open fun decrypt(reps: Int): BlackHole = blackhole.apply {
+    for(i in 0 until reps)
+      consume(client.decryptExisting(record, key))
+  }
+
+  @Benchmark
+  open fun encrypt(reps: Int): BlackHole = blackhole.apply {
+    for(i in 0 until reps)
+      consume(client.encryptRecord(recordType, RecordData(mapOf(Pair("data", bytes))), null, key))
+  }
+
+  @Benchmark
+  open fun sign(reps: Int): BlackHole = blackhole.apply {
+    val localRecord = LocalRecord(mapOf(Pair("data", bytes)), LocalMeta(client.clientId(), client.clientId(), recordType, null))
```

Comment:
Is there not a concern for object creation costs skewing the benchmark?

Reply:
The `reps` parameter is pretty high (1000+) so the object creation should be negligible.